cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -127,17 +127,29 @@ class parquet_field_int : public parquet_field {
 
   T& val;
 
+  std::string type_string()
+  {
+    if constexpr (EXPECTED_TYPE == ST_FLD_BYTE) {
+      return "byte";
+    } else if constexpr (EXPECTED_TYPE == ST_FLD_I32) {
+      return "int32";
+    } else if constexpr (EXPECTED_TYPE == ST_FLD_I64) {
+      return "int64";
+    }
+    return "unknown(EXPECTED_TYPE=" + std::to_string(EXPECTED_TYPE) + ")";
```

Comment:
could this be an "else" and then the function made constexpr?

Reply:
I don't know if it makes sense to make the function constexpr when it returns a `string`. A previous version returned a `char const*`, but CUDF_EXPECTS didn't like that.

Reply:
Guess that's a C++20 thing 😅. Someday...